cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -229,21 +229,36 @@ func (c *CStorPoolController) cStorPoolAddEvent(cStorPoolGot *apis.CStorPool) (s
 	}
 	var importPoolErr error
 	var status string
+	var importOptions pool.ImportOptions
 	cachefileFlags := []bool{true, false}
 	for _, cachefileFlag := range cachefileFlags {
-		status, importPoolErr = c.importPool(cStorPoolGot, cachefileFlag)
+		importOptions.CachefileFlag = cachefileFlag
+		status, importPoolErr = c.triggerImportPool(cStorPoolGot, &importOptions)
 		if status == string(apis.CStorPoolStatusOnline) {
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeNormal, string(common.SuccessImported), string(common.MessageResourceImported))
-			common.SyncResources.IsImported = true
 			return status, nil
 		}
 		if importPoolErr != nil {
-			glog.Errorf("failed to handle add event: import succeded with failed operations %v", importPoolErr)
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureImported), string(common.FailureImportOperations))
 			return status, importPoolErr
 		}
 	}
 
+	devIDList, err := c.getDeviceIDs(cStorPoolGot)
+	if err != nil {
+		return string(apis.CStorPoolStatusOffline), errors.Wrapf(err, "failed to get device id of disks for csp %s", cStorPoolGot.Name)
+	}
+
+	devPath := pool.GetDevPath(devIDList[0])
+	// Trigger import with dev path
+	importOptions.CachefileFlag = false
+	importOptions.DevPath = devPath
+	status, importPoolErr = c.triggerImportPool(cStorPoolGot, &importOptions)
```

Comment:
Why is this triggerImportPool called again with cachefileflag set to false ? I see import is already triggered two times dues to for loop with `true` and `false` cachefile.

Reply:
this one is done with devPath.. without cache file, pool can be imported using path in which pool devices are there..